src/test/llmtemplate_test.cpp

```diff
@@ -0,0 +1,332 @@
+    nodeResources->modelsPath = directoryPath;
+    LLMNodeResources::loadTextProcessor(nodeResources, nodeResources->modelsPath);
+
+    std::string finalPrompt = "";
+    std::string payloadBody = R"(
+        {
+            "model": "gpt",
+            "stream": false,
+            "messages": [{"role": "user", "content": "hello"}]
+        }
+    )";
+    std::string expectedOutput = "Error: Chat template not loaded correctly, so it cannot be applied";
+    ASSERT_EQ(applyChatTemplate(nodeResources->textProcessor, nodeResources->modelsPath, payloadBody, finalPrompt), false);
+    ASSERT_EQ(finalPrompt, expectedOutput);
+}
+
+TEST_F(LLMChatTemplateTest, ChatTemplateTokenizerUpperCase) {
+    std::string tokenizerJson = R"({
+    "bos_token": "</s>",
+    "eos_token": "</s>",
+    "chat_template": "{{ \"Hi, \" + messages[0]['content'] | upper }}"
+    })";
+    CreateTokenizerConfig(tokenizerJson);
+    std::shared_ptr<LLMNodeResources> nodeResources = std::make_shared<LLMNodeResources>();
+    nodeResources->modelsPath = directoryPath;
+    LLMNodeResources::loadTextProcessor(nodeResources, nodeResources->modelsPath);
+
+    std::string finalPrompt = "";
+    std::string payloadBody = R"(
+        {
+            "model": "gpt",
+            "stream": false,
+            "messages": [{"role": "user", "content": "hello"}]
+        }
+    )";
+    std::string expectedOutput = "Hi, HELLO";
+    ASSERT_EQ(applyChatTemplate(nodeResources->textProcessor, nodeResources->modelsPath, payloadBody, finalPrompt), true);
+    ASSERT_EQ(finalPrompt, expectedOutput);
+}
+
+TEST_F(LLMChatTemplateTest, ChatTemplateTokenizerTemplateException) {
+    std::string tokenizerJson = R"({
+    "bos_token": "</s>",
+    "eos_token": "</s>",
+    "chat_template": "{{ \"Hi, \" + messages[3]['content'] | upper }}"
+    })";
+    CreateTokenizerConfig(tokenizerJson);
+    std::shared_ptr<LLMNodeResources> nodeResources = std::make_shared<LLMNodeResources>();
+    nodeResources->modelsPath = directoryPath;
+    LLMNodeResources::loadTextProcessor(nodeResources, nodeResources->modelsPath);
+
+    std::string finalPrompt = "";
+    std::string payloadBody = R"(
+        {
+            "model": "gpt",
+            "stream": false,
+            "messages": [{"role": "user", "content": "hello"}]
+        }
+    )";
+    std::string expectedOutput = "list object has no element 3";
+    ASSERT_EQ(applyChatTemplate(nodeResources->textProcessor, nodeResources->modelsPath, payloadBody, finalPrompt), false);
+    ASSERT_EQ(finalPrompt, expectedOutput);
+}
+
+TEST_F(LLMChatTemplateTest, ChatTemplateTwoConfigs) {
+    std::string tokenizerJson = R"({
+    "bos_token": "</s>",
+    "eos_token": "</s>",
+    "chat_template": "{{ \"Hi, \" + messages[0]['content'] | lower }}"
+    })";
+    CreateTokenizerConfig(tokenizerJson);
+    std::string jinjaTemplate = R"( {{ "Hi, " + messages[0]['content'] | upper }} )";
```

Comment:
Let's make this template different, so we can tell which one is applied

Reply:
Its lower case and upper case, so its different.

Reply:
Oh, sorry, missed that